src/UIComponents/ColumnDetailsCategorical.jsx

```diff
@@ -0,0 +1,78 @@
+/* React component to handle showing details of categorical columns. */
+import PropTypes from "prop-types";
+import React, { Component } from "react";
+import { colors, styles } from "../constants";
+import { Bar } from "react-chartjs-2";
+
+const barData = {
+  labels: [],
+  datasets: [
+    {
+      label: "",
+      backgroundColor: colors.tealTransparent,
+      borderColor: colors.teal,
+      borderWidth: 1,
+      hoverBackgroundColor: "#59cad3",
+      hoverBorderColor: "white",
+      data: []
+    }
+  ]
+};
+
+const chartOptions = {
+  scales: {
+    yAxes: [
+      {
+        ticks: {
+          beginAtZero: true
+        }
+      }
+    ]
+  },
+  legend: { display: false },
+  maintainAspectRatio: false
+};
+
+export default class ColumnDetailsCategorical extends Component {
+  static propTypes = {
+    id: PropTypes.string,
+    uniqueOptions: PropTypes.object,
+    optionFrequencies: PropTypes.object
+  };
+
+  render() {
+    const { uniqueOptions, optionFrequencies, id } = this.props;
+
+    barData.labels = Object.values(uniqueOptions);
+    barData.datasets[0].data = barData.labels.map(option => {
+      return optionFrequencies[option];
+    });
+    barData.datasets[0].label = id;
+
+    const maxLabelsInHistogram = 5;
+
+    return (
+      <div>
+        <div style={styles.bold}>Column information:</div>
+
+        <div style={styles.barChart}>
+          {barData.labels.length <= maxLabelsInHistogram && (
+            <Bar
+              data={barData}
+              width={100}
+              height={150}
+              options={chartOptions}
+            />
```

Comment:
Maybe out of scope here, and essentially a preexisting issue that I always meant to circle back to, but it's a bit inconsistent that `ColumnDetailsCategorical` contains the bar chart component, but `ColumnDetailsNumerical` doesn't contain the scatter plot component.

Reply:
I don't think `BarChart` is to `ColumnDetailsCategorical` as `ScatterPlot` is to `ColumnDetailsNumerical`. We show a bar chart as a visualization of the values in a single categorical column. The equivalent information for a single numerical column is max/min/range. Scatterplot shows the relationship between 2 numerical columns similar to how the cross tab shows the relationship between two categorical columns. `ScatterPlot` and `CrossTab` are already fairly tidy in that they are their own components that each get the information they need directly from redux so I do plan to touch then as part of this change.